lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -25,9 +25,12 @@ def extract_facts_from_request(request)
       # in Network::HTTP::Handler will automagically deserialize the value.
       if text_facts.is_a?(Puppet::Node::Facts)
         facts = text_facts
-      else
+      elsif format == 'pson'
         # We unescape here because the corresponding code in Puppet::Configurer::FactHandler escapes
+        # PSON is deprecated, but continue to accept from older agents
```

Comment:
The ticket mentions we should issue a deprecation warning if we receive PSON. Given the likelihood of spamming the master's logs when dealing with mixed agent versions, I didn't add a deprecation warning.